src/praisonai/praisonai/cli/features/agents.py

```
@@ -148,6 +150,22 @@ def _load_tools(self, tool_names: List[str]) -> List:
+            try:
+                from praisonaiagents.hermes_filter import filter_tools_with_hermes
+                available_tools = set(tool_map.keys())
+                filtered_tools = filter_tools_with_hermes(available_tools, log_diagnostics=False)
+                
+                # Filter tool_map to only include whitelisted tools
+                tool_map = {name: func for name, func in tool_map.items() if name in filtered_tools}
+                
+            except ImportError:
+                # HERMES_ONLY_TOOLS filter not available, proceed normally
+                pass
+            except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)